### DISCUSSION TITLE
Именование профилей Ru Core

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#97] [General]
Следует принять решение о том, как визуально уточняющие профили будут отличаться от базовых, в частности, профили RuCore - от ресурсов FHIR.
Например, адрес Ru Core сейчас называется Address, и с таким наименованием включается в профиль Patient. В результате не видно, что в профиле Patient необходимо использовать российский адрес, а не базовый.

Варианты:
- принять специфические названия, например Ru_Address
- использовать URL и обеспечить его показ

**Comment by Vrodvi** [comment]
Нужны не только наименование (техническое, `name`) и URL,
ещё желательно определить формат для атрибутов "наименование для отображения" (`title`) и "id",
а также обозначить общие правила наименjвания (например - структура URL, а также случаи когда нужно более одного профиля на один тип ресурса - как обозначать их различие).

На примеры из зарубежных руководств:

### 1) Американское руководство "US Core":
```
  "id" : "us-core-patient",
  "url" : "http://hl7.org/fhir/us/core/StructureDefinition/us-core-patient",
  "name" : "USCorePatientProfile",
  "title" : "US Core Patient Profile",
```

Подход: префикс из домена (US) и названия IG (Core), затем название ресурса\типа данных, затем (не везде) слово Profile, 
написание отличается в разных атрибутах:
- в id: kebab-case - слова через тире, нижний регистр
- в name: camel-case - слова слитно, каждое слово с заглавной буквы, затем - нижний регистр (кроме аббревиатур - они все заглавными буквами)
- в title: слова через пробел, каждое слово с заглавной буквы

Когда появляется несколько профилей на один тип ресурса, то добавляется ещё один префикс с уточнением (но при этом где-то выкидывают название типа ресурса, а где-то оставляют), например:
```
  "id" : "us-core-smokingstatus",
  "url" : "http://hl7.org/fhir/us/core/StructureDefinition/us-core-smokingstatus",
  "name" : "USCoreSmokingStatusProfile",
  "title" : "US Core Smoking Status Observation Profile",
```  
  
Правила формирования URL (кроме терминологических ресурсов - там другие правила):
```
"http://hl7.org  /fhir  /us        /core  /StructureDefinition  /us-core-patient"
"[base url]      /fhir  /[domain]  /[IG]  /StructureDefinition  /[id]"
```
  
### 2) Международное руководство "IPS":
```  
  "id" : "Patient-uv-ips",
  "url" : "http://hl7.org/fhir/uv/ips/StructureDefinition/Patient-uv-ips",
  "name" : "PatientUvIps",
  "title" : "Patient (IPS)",
```
  
Подход: название ресурса\типа данных, затем суффикс из домена (UV - универсальный домен, так HL7 обозначает международные артефакты) и названия IG (IPS)
написание отличается в разных атрибутах:
- в id: kebab-case - слова через тире, первая буква заглавная, дальше - нижний регистр
- в name: camel-case - слова слитно, каждое слово с заглавной буквы, затем - нижний регистр (аббревиатуры тоже)
- в title: слова через пробел, каждое слово с заглавной буквы

Когда появляется несколько профилей на один тип ресурса, то добавляется ещё один суффикс с уточнением, например:
```
  "id" : "Observation-results-radiology-uv-ips",
  "url" : "http://hl7.org/fhir/uv/ips/StructureDefinition/Observation-results-radiology-uv-ips",
  "name" : "ObservationResultsRadiologyUvIps",
  "title" : "Observation Results: radiology (IPS)",
```

Правила формирования URL (кроме терминологических ресурсов - там другие правила):
```
"http://hl7.org  /fhir  /uv        /ips   /StructureDefinition  /Observation-results-radiology-uv-ips"
"[base url]      /fhir  /[domain]  /[IG]  /StructureDefinition  /[id]"
```

---

Кто готов высказать предложения по правилам наименования - просьба оформлять их с указанием всех озвученных выше особенностей, например:

**Подход**: префикс из домена (ru) и названия IG (core), затем название типа ресурса\типа данных, затем уточнения, в конце слово Profile.

**id**: слова через тире, нижний регистр, латинскими буквами
`"id" : "ru-core-[тип ресурса\тип данных]-[уточнения-через-тире]-profile"`
`"id" : "ru-core-patient-adult-patient-profile"`
`"id" : "ru-core-patient-newborn-patient-profile"`

**url**: [base url]      /fhir  /[domain]  /[IG]  /StructureDefinition  /[id]
`url" : "http://hl7.ru/fhir/ru/core/StructureDefinition/ru-core-[тип ресурса\тип данных]-[уточнения-через-тире]-profile`
`url" : "http://hl7.ru/fhir/ru/core/StructureDefinition/ru-core-patient-adult-patient-profile`
`url" : "http://hl7.ru/fhir/ru/core/StructureDefinition/ru-core-patient-newborn-patient-profile`

**name**: camel-case - слова слитно, каждое слово с заглавной буквы, затем - нижний регистр, латинскими буквами
`"name" : "RuCore[ТипРесурса\ТипДанных][УточненияCamelКейсом]Profile"`
`RuCorePatientAdultPatientProfile`
`RuCorePatientNewbornPatientProfile`

**title**: слова через пробел, каждое слово с заглавной буквы, аббревиатуры все заглавные, уточнения русскими буквами, можно добавлять знаки препинания
`"title" : "RU Core [Тип ресурса\Тип данных] [Уточнения Через Пробел] Profile"`
`RU Core Patient Пациент (взрослый) Profile`
`RU Core Patient Пациент (новорождённых) Profile`

**Comment by Vrodvi** [comment]
Уточнение для профилей расширений:
[Тип ресурса\Тип данных] = `Extension`, например:
`http://hl7.ru/fhir/ru/core/StructureDefinition/ru-core-extention-fias-profile`

**Comment by Vrodvi** [comment]
Уточнение для терминологических профилей:
`http://hl7.ru/fhir/ru/core/CodeSystem/[название-справочника]-ru-core`
`http://hl7.ru/fhir/ru/core/ValueSet/[название-набора-значений-один]-ru-core`
`http://hl7.ru/fhir/ru/core/ConceptMap/[название-набора-значений-один]-[название-набора-значений-два]-ru-core`

**Comment by VarvaraSemenova** [comment]
Поддерживаю описанный @Vrodvi формат, единственное, предлагаю не дублировать название ресурса в уточнении профиля (например, вместо "id" : "ru-core-patient-adult-patient-profile" писать "id" : "ru-core-patient-adult-profile").  Базовый профиль на пациента будет называться так:  "id" : "ru-core-patient-profile"

Подход: префикс из домена (ru) и названия IG (core), затем название типа ресурса\типа данных, затем уточнения, в конце слово Profile.

id: слова через тире, нижний регистр, латинскими буквами
"id" : "ru-core-[тип ресурса\тип данных]-[уточнения-через-тире]-profile"
"id" : "ru-core-patient-adult-profile"
"id" : "ru-core-patient-newborn-profile"

url: [base url] /fhir /[domain] /[IG] /StructureDefinition /[id]
url" : "http://hl7.ru/fhir/ru/core/StructureDefinition/ru-core-[тип ресурса\тип данных]-[уточнения-через-тире]-profile
url" : "http://hl7.ru/fhir/ru/core/StructureDefinition/ru-core-patient-adult-profile
url" : "http://hl7.ru/fhir/ru/core/StructureDefinition/ru-core-patient-newborn-profile

name: camel-case - слова слитно, каждое слово с заглавной буквы, затем - нижний регистр, латинскими буквами
"name" : "RuCore[ТипРесурса\ТипДанных][УточненияCamelКейсом]Profile"
RuCorePatientAdultProfile
RuCorePatientNewbornProfile

Вопрос: какие варианты профилей RU-, но не core? Какая тогда схема замены слова 'core'?

**Reply by Vrodvi** [reply]
После core-профилей будут специфические профили под конкретные задачи, например:

- профиль `frmo` для учёта специфики Федерального Регистра Медицинских Организаций, например:
"url": "http://hl7.ru/fhir/ru/frmo/StructureDefinition/ru-frmo-organization-legal-entity-profile" - для Организации-юридического лица
"url": "http://hl7.ru/fhir/ru/frmo/StructureDefinition/ru-frmo-organization-branch-profile" - для филиала
"url": "http://hl7.ru/fhir/ru/frmo/StructureDefinition/ru-frmo-organization-division-profile" - для подразделения

- профиль `semd` для учёта специфики Структурированных Электронных Медицинских Документов, например:
"url": "http://hl7.ru/fhir/ru/semd/StructureDefinition/ru-semd-composition-discharge-epicrisis-profile" - для СЭМДа "Эпикриз в стационаре выписной"

и т.п.

**Reply by niquola** [reply]
В целом согласен - но немного убрал бы реданданси:

`http://fhir.ru/<имя-ig:core>/StructureDefinition/<resource-type>-<proile-name> без "-profile"`

```
url: "http://fhir.ru/core/StructureDefinition/organization-legal-entity"
url: "http://fhir.ru/frmo/StructureDefinition/organization-division"
url: "http://fhir.ru/semd/StructureDefinition/composition-discharge-epicrisis"
```

**Reply by niquola** [reply]
как вариант можно ввести сокращения sd - StructureDefinition; vs - Valueset - тогда будет еще короче:
```
url: "http://fhir.ru/core/sd/organization-legal-entity"
url: "http://fhir.ru/frmo/sd/organization-division"
url: "http://fhir.ru/semd/sd/composition-discharge-epicrisis"
```

**Comment by Vrodvi** [comment]
_**Дублирую как отдельный комментарий, т.к. в предыдущей цепочке комментариев не видно.**_

Итого, на данный момент с учётом всех предложений получаем:

**базовый url**: `http://fhir.ru`

**Подход**: затем название IG (core), затем название типа (тип ресурса, тип данных, справочник, набор значений, расширение), затем уточнения

---

Конкретно:

---

**id**: слова через тире, нижний регистр, латинскими буквами
`"id" : "core-[тип]-[уточнения-через-тире]"`
`[тип]` - тип ресурса, тип данных, ex для extension, cs для codesystem, vs для valueset, cm для conceptmap (плюс буква -x- между названиями маппируемых наборов значений)

ресурс
`"id" : "core-patient-adult"`
`"id" : "core-patient-newborn"`
тип данных
`"id" : "core-address"`
расширение
`"id" : "core-ex-fias"`
терминологии
`"id" : "core-cs-nsi-sex"`
`"id" : "core-vs-nsi-sex"`
`"id" : "core-cm-core-vs-nsi-sex-x-administrative-gender"`

---

**url**: "[base url]      /[IG] /[тип артефакта]/[id]"
`"url" :   "http://fhir.ru/core/[тип артефакта]/core-[тип]-[уточнения-через-тире]"`
`[тип артефакта]` - sd для StructureDefinition (в том числе для типов данных и для extension), cs для codesystem, vs для valueset, cm для conceptmap (плюс буква -x- между названиями маппируемых наборов значений)
`[тип]` - тип ресурса, тип данных, ex для extension, cs для codesystem, vs для valueset, cm для conceptmap (плюс буква -x- между названиями маппируемых наборов значений)

ресурс
`"url" : "http://fhir.ru/core/sd/core-patient-adult"`
`"url" : "http://fhir.ru/core/sd/core-patient-newborn"`
тип данных
`"url" : "http://fhir.ru/core/sd/core-address"`
расширение
`"url" : "http://fhir.ru/core/sd/core-ex-fias-profile"`
терминологии
`"url" : "http://fhir.ru/core/cs/core-cs-nsi-sex"`
`"url" : "http://fhir.ru/core/vs/core-vs-nsi-sex"`
`"url" : "http://fhir.ru/core/cm/core-cm-core-vs-nsi-sex-x-administrative-gender"`

---

**name**: camel-case - слова слитно, каждое слово с заглавной буквы, затем - нижний регистр, латинскими буквами
`"name" : "Core[Тип][УточненияCamelКейсом]"`
`[Тип]` - Тип ресурса, Тип данных, Ex для extension, Cs для codesystem, Vs для valueset, Cm для conceptmap (плюс буква X между названиями маппируемых наборов значений)

ресурс
`"name" : "CorePatientAdult"`
`"name" : "CorePatientNewborn"`
тип данных
`"name" : "CoreAddress"`
расширение
`"name" : "CoreExFias"`
терминологии
`"name" : "CoreCsNsiSex"`
`"name" : "CoreVsNsiSex"`
`"name" : "CoreCmCoreVsNsiSexXAdministrativeGender"`

---

**title**: слова через пробел, каждое слово с заглавной буквы, аббревиатуры все заглавные, уточнения русскими буквами, можно добавлять знаки препинания
`"title" : "Core [Тип] [Уточнения Через Пробел]"`
`[Тип]` - Тип ресурса, Тип данных, Extension, CodeSystem, ValueSet, ConceptMap (плюс буква X между названиями маппируемых наборов значений)

ресурс
`"title" : "Core Patient Пациент (взрослый)"`
`"title" : "Core Patient Пациент (новорождённых)"`
тип данных
`"title" : "Core Address Адрес"`
расширение
`"title" : "Core Extension ФИАС"`
терминологии
`"title" : "Core CodeSystem Пол пациента (ФНСИ)"`
`"title" : "Core ValueSet Пол пациента (ФНСИ)"`
`"title" : "Core ConceptMap Core ValueSet Пол пациента (ФНСИ) X AdministrativeGender"`
